flux_led/__main__.py

```diff
@@ -574,32 +576,68 @@ def _determineMode(self, ww_level, pattern_code):
                 mode = "color"
         elif pattern_code == 0x60:
             mode = "custom"
+        elif pattern_code == 0x41:
+            mode = "color"
         elif PresetPattern.valid(pattern_code):
             mode = "preset"
         elif BuiltInTimer.valid(pattern_code):
             mode = BuiltInTimer.valtostr(pattern_code)
 
         return mode
 
-    def update_state(self, retry=2):
+    def query_state(self, retry=2, led_type = None):
+
+        # default values
         msg = bytearray([0x81, 0x8a, 0x8b])
+        msg_len = 14
+
+        if self.protocol == 'LEDnet_wifi370' or led_type == 'LEDnet_wifi370':
+            msg =  bytearray([0xef, 0x01, 0x77])
+            msg_len = 11
+            led_type = 'LEDnet_wifi370'
+            self._use_csum = False
         try:
             self._send_msg(msg)
-            rx = self._read_msg(14)
+            rx = self._read_msg(msg_len)
```

Comment:
Could you explain how this is a problem? 

As I understand it, the original protocol sends out a query  bytearray([0xef, 0x01, 0x77]) and expects 11 bytes as a response.
the newer protocols all send out a query bytearray([0x81, 0x8a, 0x8b]) and expects 14 bytes as a response. 

if you don't specify that the original protocol should be used, the default (new) bytearray is sent and a new 14 byte response is expected.  

btw, I think this part
```
        print('rx length', len(rx), 'led_type = ', led_type)  
        if len(rx) == 11 and led_type == 'LEDENET_ORIGINAL':
            self.protocol = 'LEDENET_ORIGINAL'
```
is not needed as that check is done in update_state. Will leave it out.